cli/testflinger_cli/client.py

```diff
@@ -86,6 +87,30 @@ def put(self, uri_frag, data, timeout=15):
             raise HTTPError(req.status_code)
         return req.text
 
+    def put_file(self, uri_frag: str, path: Path, timeout: float) -> str:
+        """Stream a file to the server using a POST request
+        :param uri_frag:
+            endpoint for the POST request
+        :return:
+            String containing the response from the server
+        """
+        uri = urllib.parse.urljoin(self.server, uri_frag)
+        with open(path, "rb") as file:
+            try:
+                files = {"file": (path.name, file)}
+                response = requests.post(uri, files=files, timeout=timeout)
+            except requests.exceptions.ConnectTimeout:
+                logger.error(
+                    "Timout while trying to communicate with the server."
+                )
+                sys.exit(1)
+            except requests.exceptions.ConnectionError:
+                logger.error("Unable to communicate with specified server.")
+                sys.exit(1)
```

Comment:
You use `raise SystemExit()` above, but `logger.error()` + `sys.exit(1)` here. Not sure which one is preferred by the testflinger project, but we should be consistent with one or the other if possible.